### PR TITLE
qa: valgrind: disable AVX

### PR DESCRIPTION
**Pull request by yehudasa**
Fixes: https://tracker.ceph.com/issues/44362

Signed-off-by: Yehuda Sadeh <yehuda@redhat.com>

## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard backend`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

**Comment by yehudasa** [comment]
@liewegas @rzarzynski http://pulpito.ceph.com/yehudasa-2020-03-16_03:56:13-rados:verify-wip-valgrind-no-avx-distro-basic-smithi/

**Comment by rzarzynski** [comment]
@yehudasa: if we decide to merge this, we could remove the whitelisting rules as well:

```
# "Conditional jump or move depends on uninitialised value(s)" in OpenSSL
# while using aes-128-gcm with AES-NI enabled. Not observed while running
# with `OPENSSL_ia32cap="~0x200000200000000"`.
{
   uninitialised gcm.Xi in aes-128-gcm with AES-NI for msgr, part 1
   Memcheck:Cond
   ...
   fun:EVP_DecryptFinal_ex
   fun:_ZN4ceph6crypto6onwire25AES128GCM_OnWireRxHandler34authenticated_decrypt_update_finalEONS_6buffer7v*4listEj
   fun:_ZN10ProtocolV231handle_read_frame_epilogue_mainEOSt10unique_ptrIN4ceph6buffer7v*8ptr_nodeENS4_8disposerEEi
   fun:_ZN10ProtocolV216run_continuationER2CtIS_E
   ...
   fun:_ZN15AsyncConnection7processEv
   fun:_ZN11EventCenter14process_eventsEjPNSt6chrono8durationImSt5ratioILl1ELl1000000000EEEE
   ...
}

{
   uninitialised gcm.Xi in aes-128-gcm with AES-NI for msgr, part 2
   Memcheck:Cond
   fun:_ZN4ceph6crypto6onwire25AES128GCM_OnWireRxHandler34authenticated_decrypt_update_finalEONS_6buffer7v*4listEj
   fun:_ZN10ProtocolV231handle_read_frame_epilogue_mainEOSt10unique_ptrIN4ceph6buffer7v*8ptr_nodeENS4_8disposerEEi
   fun:_ZN10ProtocolV216run_continuationER2CtIS_E
   ...
   fun:_ZN11EventCenter14process_eventsEjPNSt6chrono8durationImSt5ratioILl1ELl1000000000EEEE
   ...
}
```

File: qa/suites/fs/verify/validater/valgrind.yaml [hunk 1]
```diff
@@ -28,3 +28,6 @@ overrides:
   ceph-fuse:
     client.0:
       valgrind: [--tool=memcheck, --leak-check=full, --show-reachable=yes]
+  workunit:
+    env:
+      OPENSSL_ia32cap: "~0x1000000000000000"
```
Comment (rzarzynski) [inline qa/suites/fs/verify/validater/valgrind.yaml]: How about putting a comment with link to the tracker?

**Comment by liewegas** [comment]
I see lots of other instances of valgrind:
```
../qa/suites/fs/verify/validater/valgrind.yaml:    valgrind:
../qa/suites/fs/verify/validater/valgrind.yaml:      valgrind: [--tool=memcheck, --leak-check=full, --show-reachable=yes]
../qa/suites/rados/valgrind-leaks/1-start.yaml:    valgrind:
../qa/suites/rados/valgrind-leaks/2-inject-leak/mon.yaml:    expect_valgrind_errors: true
../qa/suites/rados/valgrind-leaks/2-inject-leak/osd.yaml:    expect_valgrind_errors: true
../qa/suites/rados/verify/validater/valgrind.yaml:# valgrind is slow.. we might get PGs stuck peering etc
../qa/suites/rados/verify/validater/valgrind.yaml:# mons sometimes are left off of initial quorum due to valgrind slowness.  ok to whitelist here because we'll still catch an actual crash due to the core
../qa/suites/rados/verify/validater/valgrind.yaml:    valgrind:
../qa/suites/rbd/valgrind/validator/memcheck.yaml:    valgrind: ["--tool=memcheck"]
../qa/suites/rgw/multisite/tasks/test_multi.yaml:# ubuntu and no valgrind until we migrate test to py3
../qa/suites/rgw/multisite/tasks/test_multi.yaml:#      valgrind: [--tool=memcheck, --max-threads=1024] # http://tracker.ceph.com/issues/25214
../qa/suites/rgw/multisite/tasks/test_multi.yaml:#      valgrind: [--tool=memcheck, --max-threads=1024]
../qa/suites/rgw/multisite/tasks/test_multi.yaml:#      valgrind: [--tool=memcheck, --max-threads=1024]
../qa/suites/rgw/multisite/tasks/test_multi.yaml:#      valgrind: [--tool=memcheck, --max-threads=1024]
../qa/suites/rgw/multisite/valgrind.yaml:# ubuntu and no valgrind until we migrate test to py3
../qa/suites/rgw/multisite/valgrind.yaml:#    valgrind:
../qa/suites/rgw/verify/tasks/0-install.yaml:#      valgrind: [--tool=memcheck, --max-threads=1024] # http://tracker.ceph.com/issues/25214
../qa/suites/rgw/verify/validater/valgrind.yaml:#    valgrind:
../qa/suites/rgw/website/overrides.yaml:#        valgrind: [--tool=memcheck, --max-threads=1024] # http://tracker.ceph.com/issues/25214
../qa/suites/rgw/website/overrides.yaml:#        valgrind: [--tool=memcheck, --max-threads=1024] # http://tracker.ceph.com/issues/25214
```
they should all get this i'm assuming?

**Comment by yehudasa** [comment]
@liewegas @rzarzynski  updated

File: qa/suites/fs/verify/validater/valgrind.yaml (hunk 1 above)
Comment (yehudasa) [reply qa/suites/fs/verify/validater/valgrind.yaml]: done

**Comment by yehudasa** [comment]
@liewegas rebased

**Comment by liewegas** [comment]
I still see valgrind complaining here:
 /a/sage-2020-03-16_23:53:52-rados-wip-sage-testing-2020-03-16-1740-distro-basic-smithi/4861121

```
  <kind>SyscallParam</kind>
  <what>Syscall param sendmsg(msg.msg_iov[1]) points to uninitialised byte(s)</what>
  <stack>
    <frame>
      <ip>0xC250AF7</ip>
      <obj>/usr/lib64/libpthread-2.28.so</obj>
      <fn>sendmsg</fn>
    </frame>
    <frame>
      <ip>0x10C9AFF</ip>
      <obj>/usr/bin/ceph-osd</obj>
      <fn>do_sendmsg</fn>
      <dir>/usr/src/debug/ceph-15.1.1-85.g6b18e0a.el8.x86_64/src/msg/async</dir>
      <file>PosixStack.cc</file>
      <line>80</line>
    </frame>
    <frame>
      <ip>0x10C9AFF</ip>
      <obj>/usr/bin/ceph-osd</obj>
      <fn>PosixConnectedSocketImpl::send(ceph::buffer::v15_2_0::list&amp;, bool)</fn>
      <dir>/usr/src/debug/ceph-15.1.1-85.g6b18e0a.el8.x86_64/src/msg/async</dir>
      <file>PosixStack.cc</file>
      <line>129</line>
    </frame>
    <frame>
      <ip>0x107FBA0</ip>
      <obj>/usr/bin/ceph-osd</obj>
      <fn>send</fn>
      <dir>/usr/src/debug/ceph-15.1.1-85.g6b18e0a.el8.x86_64/src/msg/async</dir>
      <file>Stack.h</file>
      <line>100</line>
    </frame>
    <frame>
      <ip>0x107FBA0</ip>
      <obj>/usr/bin/ceph-osd</obj>
      <fn>AsyncConnection::_try_send(bool)</fn>
      <dir>/usr/src/debug/ceph-15.1.1-85.g6b18e0a.el8.x86_64/src/msg/async</dir>
      <file>AsyncConnection.cc</file>
      <line>330</line>
    </frame>
    <frame>
      <ip>0x10801CA</ip>
      <obj>/usr/bin/ceph-osd</obj>
      <fn>AsyncConnection::write(ceph::buffer::v15_2_0::list&amp;, std::function&lt;void (long)&gt;, bool)</fn>
      <dir>/usr/src/debug/ceph-15.1.1-85.g6b18e0a.el8.x86_64/src/msg/async</dir>
      <file>AsyncConnection.cc</file>
      <line>309</line>
    </frame>
    <frame>
      <ip>0x10AB0C5</ip>
      <obj>/usr/bin/ceph-osd</obj>
      <fn>ProtocolV2::write(std::__cxx11::basic_string&lt;char, std::char_traits&lt;char&gt;, std::allocator&lt;char&gt; &gt; const&amp;, CtFun&lt;ProtocolV2&gt;&amp;, ceph::buffer::v15_2_0::list&amp;)</fn>
      <dir>/usr/src/debug/ceph-15.1.1-85.g6b18e0a.el8.x86_64/src/msg/async</dir>
      <file>ProtocolV2.cc</file>
      <line>786</line>
    </frame>
    <frame>
      <ip>0x10C7B92</ip>
      <obj>/usr/bin/ceph-osd</obj>
      <fn>Ct&lt;ProtocolV2&gt;* ProtocolV2::write&lt;ceph::msgr::v2::AuthSignatureFrame&gt;(std::__cxx11::basic_string&lt;char, std::char_traits&lt;char&gt;, std::allocator&lt;char&gt; &gt; const&amp;, CtFun&lt;ProtocolV2&gt;&amp;, ceph::msgr::v2::AuthSignatureFrame&amp;)</fn>
      <dir>/usr/src/debug/ceph-15.1.1-85.g6b18e0a.el8.x86_64/src/msg/async</dir>
      <file>ProtocolV2.cc</file>
      <line>773</line>
    </frame>
    <frame>
      <ip>0x10B5CD8</ip>
      <obj>/usr/bin/ceph-osd</obj>
      <fn>ProtocolV2::handle_auth_done(ceph::buffer::v15_2_0::list&amp;)</fn>
      <dir>/usr/src/debug/ceph-15.1.1-85.g6b18e0a.el8.x86_64/src/msg/async</dir>
      <file>ProtocolV2.cc</file>
      <line>1882</line>
    </frame>
```

**Comment by yehudasa** [comment]
@liewegas as far as I can understand, the workunit task is not being run there in the specific test so env isn't being set.

**Comment by liewegas** [comment]
> @liewegas as far as I can understand, the workunit task is not being run there in the specific test so env isn't being set.

Ah.. yeah I think this doesn't address it then.  I think the fix is to adjust get_valgrind_args in teuthology.git/teuthology/misc.py

**Comment by liewegas** [comment]
https://github.com/ceph/teuthology/commit/eb6be8d5d4ee560f1c790898b03393fa3bf31656
?

**Comment by liewegas** [comment]
https://github.com/ceph/teuthology/pull/1425

**Pull request closed without merging**